internal/bootstrap/bootstrap_config.go

```diff
@@ -795,6 +805,60 @@ func (c *BootstrapConfig) generateListenerConfig(args *BootstrapTplArgs, bindAdd
 	return nil
 }
 
+// appendHCPMetricsConfig generates config to enable a socket at path: <hcpMetricsBindSocketDir>/<namespace>_<proxy_id>.sock
+// or <hcpMetricsBindSocketDir>/<proxy_id>.sock, if namespace is empty.
+func appendHCPMetricsConfig(args *BootstrapTplArgs, hcpMetricsBindSocketDir string) {
+	dir := hcpMetricsBindSocketDir
+	if !strings.HasSuffix(dir, "/") {
+		dir += "/"
+	}
+
+	path := fmt.Sprintf("%s%s_%s.sock",dir,args.Namespace,args.ProxyID)
```

Comment:
After testing locally with @freddygv , looks like the Namespace is always `default`, so we remove the if statement.  

See the equivalent in `consul`: https://github.com/hashicorp/consul/pull/16511/files#diff-3d65fe35c13bd0d74e9cabcee963eea26abcafb433b75454c7bfd697f1f451f4R820